db/version_set.cc

```diff
@@ -2724,6 +2724,11 @@ void VersionStorageInfo::ComputeExpiredTtlFiles(
     const ImmutableOptions& ioptions, const uint64_t ttl) {
   assert(ttl > 0);
 
+  // If we schedule compaction after TTL is reached, TTL is always slightly
+  // violated, which might be confusing to users. Instead, we make a
+  // scheduling TTL slightly ahead of time, but no more than a day.
+  uint64_t schedule_ttl = ttl - std::min(uint64_t{24 * 60 * 60}, ttl / 32);
```

Comment:
Personally, I feel this is more confusing that just reduces a little bit ttl time, but still no guarantee. I would rather just updating the doc to let the user to do that.